drafts/taking-a-byte-of-the-carbon-cake.md

```diff
@@ -0,0 +1,291 @@
+Total system carbon footprint for 3 environments = 175 * 3 = 525 KgCO2/year
+
+```
+![Figure: Relative Carbon Footprints, showing our Example Micro-service Cloud System](https://cdn.hashnode.com/res/hashnode/image/upload/v1730817205560/Y1ZfUGOdl.jpg?auto=format)
+*Figure: Relative Carbon Footprints, including our Example Micro-service Cloud System*
+
+Let’s place our micro-services cloud system on the relative carbon footprint line. At 175 KgCO2 it would be somewhere in the middle between running a lawn mower and owning a cat. 
+But of course any realistic software system will likely have 2-3 similar environments, with varied configurations, plus services such as gateways, load balances, databases etc...
+These services may be PAYG shared, rather than dedicated to this system, but nevertheless need to be accounted. 
+
+In practice, it is likely that this carbon cost will be a multiple, let’s say 3x175 = 522 KgCO2/year for our example. This system is small scale in comparison to many brand-name e-commerce retail sites or banking systems, yet already close to the cost of a flight from London to Malaga at 620 KgCO2.
+
+
+## Green washing?
+
+Any article on carbon footprint is almost obliged to mention 'green washing', the subtle art of hiding the truth whilst appearing transparent on climate change and carbon footprints.
+
+Amazon has a target of 100% green by their own measurement. 
+That does NOT mean that your system will be carbon-friendly on AWS, it simply means that AWS are accounting for the offsets. 
+In regions where they are truly powered by green energy, that same power could be used elsewhere other than their data centre, and what incentive is there to decrease their (or our) usage?   
+
+Ultimately, the green software foundation's [GSF energy efficiency policy](https://learn.greensoftware.foundation/energy-efficiency) suggests we should aim to use less energy, and that has to be the end goal. Green data centres do not equal carbon-free environments in which to run our software.
+
+Other considerations: we have ignored the carbon cost of data transfers, which are small but can accumulate. 
+That said there are few metrics available from cloud providers to capture transfers within their network, and thus within your system’s network.
+
+
+## What can we do?  What 'byte' can we take?
+
+So to reduce the carbon footprint of the software we create, what ‘byte’ can we take of the carbon cake?
+
+First, there are some excellent best practices available on sites such as [Patterns, Green Software Foundation](https://patterns.greensoftware.foundation/)
+and I recommend a read through the patterns. Amazon Cloud (AWS) also have valuable guidance in their AWS Well-architected Sustainability.  Google Cloud (GCP) also has good credentials for transparency in how they calculate the carbon footprint of their services.
+
+With an engineering mindset, it’s tempting to focus on the challenge of improvements to compute workloads; 
```

Comment:
interesting thoughts in these three paragraphs.

In the flow of the article I think they bring a bit friction into it. What about leaving this discussion about the key pattern just away and skip to the 10 measures?

Reply:
the suggestion is to condense lines 170-181 into a single paragraph by removing the discussion about the key patterns (which mainly leaves the introduction of the next chapter).

Reply:
I agree, I've reworked this section and reduced the content